src/v/cloud_storage/remote_partition.cc

```diff
@@ -753,6 +753,15 @@ remote_partition::timequery(storage::timequery_config cfg) {
     }
 }
 
+bool remote_partition::bounds_timestamp(model::timestamp t) const {
```

Comment:
nit: I'd give this a different name. `t` may be smaller than the min timestamp in the cloud log, in which case we'd return true even if `t` is not in the cloud log.